gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/GenerateMetricSchemaTask.java

```diff
@@ -40,8 +42,16 @@
 
 @CacheableTask
 public abstract class GenerateMetricSchemaTask extends DefaultTask {
-    private final Property<String> libraryName =
-            getProject().getObjects().property(String.class).value(defaultLibraryName());
+    private final Property<String> libraryName = getProject()
+            .getObjects()
+            .property(String.class)
+            .convention(getProject().provider(this::defaultLibraryName));
```

Comment:
I think using a convention is more accurate here than setting `value` explicitly. I updated this to evaluate lazily instead of immediately scraping the version when this task is created, because that seemed safer.

Reply:
afaik the only difference between `convention` and `value` is that if someone later comes and does `property.set(null)`, the convention would be used rather than the property being empty. This barely ever (never?) happens so it doesn't really matter - and sometimes you actually want to be able to set the property to empty (depends on usage - probably not in this case).